crates/sui-json-rpc/src/read_api.rs

```diff
@@ -746,6 +746,41 @@ impl ReadApiServer for ReadApi {
     async fn get_checkpoint(&self, id: CheckpointId) -> RpcResult<Checkpoint> {
         Ok(self.get_checkpoint_internal(id)?)
     }
+
+    async fn get_checkpoints(
+        &self,
+        cursor: Option<CheckpointSequenceNumber>,
+        limit: Option<usize>,
+        descending_order: bool,
+    ) -> RpcResult<CheckpointPage> {
+        let limit = cap_page_limit(limit);
+
+        let checkpoints = self
```

Comment:
can you follow the pagination logic in https://github.com/MystenLabs/sui/blob/main/crates/sui-json-rpc/src/read_api.rs#L739-L763 ? we should not need to use `start_index` anymore after getting the paginated result from `get_checkpoints`